### DISCUSSION TITLE
in 8.0.0-beta.9 can header in createDataColumn support internationalization with i18next?

### DISCUSSION THREAD

**Discussion by rabindranathfv** [TanStack/table#4073] [Q&A]
i am trying to generate a cell with dinamic and internationalization headers as the example. remenber use t based on i18next implementation like this 

`import { useTranslation } from "react-i18next";`

`const { t } = useTranslation();`

`tablePartner.createDataColumn('example', {
        id: 'example',
        header: t("table.headerExample")
 })
`

important i also have file.json with the right structure with
` "table": {
     "headerExample": "some translation"
}
`

react-table 8.0.0-beta.9
typescript 4.7.4

**Comment by mdrahiem** [comment]
I am using `8.10.7` and yes it does support.
My implementation is like below.

Inside the table component,

`const { t } = useTranslation('common')`
`const table = useReactTable({
        columns: getColumns(t),
        data: countryListComponent,
        getCoreRowModel: getCoreRowModel(),
        getSortedRowModel: getSortedRowModel(),
    })`

And then to create columns,
`const getColumns = (t: TFunction<'common', undefined>) => {
    return [
        countryColumnHelper.accessor('code', {
            header: (header) =>{t('country', { count: 2 })},
            enableSorting: true,
            cell: (cell) => (
                {cell.getValue()}
            ),
        }),
        ...
        rest of the columns
        `

**Reply by lennerd** [reply]
This will rerender the whole table each time because the columns are not a stable reference (get recreated on every render).
